.circleci/config.yml

```diff
@@ -0,0 +1,91 @@
+# These environment variables must be set in CircleCI UI
+#
+# DOCKERHUB_REPO - docker hub repo, format: <username>/<repo>
+# DOCKER_EMAIL   - login info for docker hub
+# DOCKER_USER
+# DOCKER_PASS
+# DOCKER_DEPLOY - set to true to deploy to Dockerhub
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # base image
+      - image: ubuntu:16.04
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: Install essential packages
+          command: |
+            apt-get update
+            apt-get install -y ca-certificates curl build-essential make git
+
+      - run:
+          name: Create version.json
+          command: |
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+      - store_artifacts:
+          path: version.json
+      # FIXME: should use an image w/ docker installed by default
+      - run:
+          name: Install Docker
+          command: |
+            set -x
+            VER="17.12.1-ce"
+            curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$VER.tgz
+            tar -xz -C /tmp -f /tmp/docker-$VER.tgz
+            mv /tmp/docker/* /usr/bin
+      - run:
+          name: Build deployment container image
+          command: docker build -t app:build .
+      - run:
+          name: Create Docker image shasum256
+          command: |
+            docker images --no-trunc |
+            awk '/^app/ {print $3}' |
+            tee docker-image-shasum256.txt
+      - store_artifacts:
+          path: docker-image-shasum256.txt
+      - run:
+          name: Test flake8
+          command: docker run app:build flake8 syncserver
+      - run:
+          name: Test nose
+          command: docker run app:build nosetests syncstorage.tests
+      - run:
+          name: Push to Dockerhub
+          command: |
+            # set DOCKER_DEPLOY=true in Circle UI to push to Dockerhub
+            DOCKER_DEPLOY="${DOCKER_DEPLOY:-false}"
+            if [ "${CIRCLE_BRANCH}" == "master" ]; then
+              bin/ci/deploy-dockerhub.sh latest
+            fi
+            if [[ "${CIRCLE_BRANCH}" == feature* ]] || [[ "${CIRCLE_BRANCH}" == dockerpush* ]]; then
```

Comment:
We should re-file this pr as one of these branches to test to make sure it actually works

Reply:
I don't think I can do this without write access to the repo. If you created a branch off `master`, then I can refile against that branch.

Reply:
I made a copy of this in a feature branch here: https://github.com/mozilla-services/syncserver/pull/120

Reply:
It build successfully over in #120, but it pushed to dockerhub as "latest" rather than as a tag named after the feature-branch.  I think the line below this should be ` bin/ci/deploy-dockerhub.sh "${CIRCLE_BRANCH}"` rather than ` bin/ci/deploy-dockerhub.sh latest`.

Reply:
You're right, I've fixed that